### PR TITLE
AB#35824 feat: removed react-hot-toast, added onSigningComplete 

### PR DESCRIPTION
**Pull request by im-nmashukov**
[AB#35824](https://dev.azure.com/Interfirst-DevOps/ec67a686-9dcc-441d-a2a0-3cc4609082d2/_workitems/edit/35824)
removed react-hot-toast, 

added: 
- onSigningComplete - triggered when Signing succeeds
- showSuccessMessage - provide to showing toast logic inside
- showErrorMessage - provide with showing toast logic inside
- providedSignerName - the name of Signer that will be used by default on non-preview Mode

### CODE REVIEW

**Review by im-termakov — APPROVED** [review]

File: src/components/FormPreviewContent/FormPreviewContent.js [hunk 1]
```diff
@@ -107,6 +120,18 @@ const FormPreviewContent = ({ signingRequestId, getToken, getDomainName }) => {
     }
   }, [data, elements]);
 
+  useEffect(() => {
+    if (!viewer && providedSignerName) {
+      setViewer(providedSignerName ? { value: providedSignerName } : null);
+    }
+  }, [viewer, providedSignerName]);
```
Comment (im-rremniov) [inline src/components/FormPreviewContent/FormPreviewContent.js:127]: [req] move it ti PreviewAsAutocomplete

File: src/components/FormPreviewContent/FormPreviewContent.js [hunk 2]
```diff
@@ -107,6 +120,18 @@ const FormPreviewContent = ({ signingRequestId, getToken, getDomainName }) => {
...
+
+  useEffect(() => {
+    if (isSigningFinished) {
+      onSigningComplete();
+    }
+  }, [isSigningFinished, onSigningComplete]);
```
Comment (im-rremniov) [inline src/components/FormPreviewContent/FormPreviewContent.js:133]: [req] place it in  submit's `then`

File: src/components/FormPreviewContent/FormPreviewContent.js [hunk 3]
```diff
@@ -23,7 +20,15 @@ import SigningFields from '../SigningFields';
...
 
-const FormPreviewContent = ({ signingRequestId, getToken, getDomainName }) => {
+const FormPreviewContent = ({
+  providedSignerName,
+  signingRequestId,
+  getToken,
+  getDomainName,
+  showErrorMessage,
+  showSuccessMessage,
+  onSigningComplete,
```
Comment (im-rremniov) [inline src/components/FormPreviewContent/FormPreviewContent.js:30]: [req] extend `FormPreviewContent.defaultProps`

File: src/components/FormPreviewContent/FormPreviewContent.js [hunk 4]
```diff
@@ -174,6 +199,7 @@ const FormPreviewContent = ({ signingRequestId, getToken, getDomainName }) => {
             <DivTop>
               <PreSignPanel
                 isLoading={isLoading}
+                isPreviewMode={!providedSignerName}
```
Comment (im-rremniov) [inline src/components/FormPreviewContent/FormPreviewContent.js:202]: [req] it is always `preview` mode, rename `isPreviewMode` var. is there way to put as init state signerName `const [viewer, setViewer] = useState(null);` here?

File: src/components/FormPreviewContent/FormPreviewContent.js [hunk 5]
```diff
@@ -23,7 +20,15 @@ import SigningFields from '../SigningFields';
...
 import PreSignPanel from './PreSignPanel';
 
-const FormPreviewContent = ({ signingRequestId, getToken, getDomainName }) => {
+const FormPreviewContent = ({
+  providedSignerName,
+  signingRequestId,
+  getToken,
+  getDomainName,
+  showErrorMessage,
+  showSuccessMessage,
```
Comment (im-rremniov) [inline src/components/FormPreviewContent/FormPreviewContent.js:29]: [req] guess need to use `context`

**Review by Crainov — APPROVED** [review]

File: src/components/FormPreviewContent/FormPreviewContent.js (hunk 2 above)
Comment (im-nmashukov) [reply src/components/FormPreviewContent/FormPreviewContent.js]: fixed https://github.com/Interfirst/Skynet.Signing.UI/pull/2/commits/5ee5c74463808d2346025d6d8f49dc2025de744e

File: src/components/FormPreviewContent/FormPreviewContent.js (hunk 3 above)
Comment (im-nmashukov) [reply src/components/FormPreviewContent/FormPreviewContent.js:30]: fixed https://github.com/Interfirst/Skynet.Signing.UI/pull/2/commits/5ee5c74463808d2346025d6d8f49dc2025de744e

File: src/components/FormPreviewContent/FormPreviewContent.js (hunk 4 above)
Comment (im-nmashukov) [reply src/components/FormPreviewContent/FormPreviewContent.js]: renamed https://github.com/Interfirst/Skynet.Signing.UI/pull/2/commits/5ee5c74463808d2346025d6d8f49dc2025de744e

File: src/components/FormPreviewContent/FormPreviewContent.js (hunk 5 above)
Comment (im-nmashukov) [reply src/components/FormPreviewContent/FormPreviewContent.js:29]: discussed, will keep it as it is for now

File: src/components/FormPreviewContent/FormPreviewContent.js (hunk 1 above)
Comment (im-nmashukov) [reply src/components/FormPreviewContent/FormPreviewContent.js]: done https://github.com/Interfirst/Skynet.Signing.UI/pull/2/commits/d00294adde82e1c9b02bb9ecb466f30d249d11ee

**Review by im-rremniov — APPROVED** [review]

**Pull request merged**